src/sync/deployment.test.ts

```diff
@@ -30,6 +37,221 @@ describe("sync/deployments", () => {
+				});
+				return clone;
+			});
+		};
+
+		const nockFetchingDeploymentgPagesGraphQL = (cursor, deployments) => {
+			githubNock.post("/graphql", { query: getDeploymentsQuery, variables: { owner: repoSyncState.repoOwner, repo: repoSyncState.repoName, per_page: PAGE_SIZE__TWO_ITEMS, cursor } })
+				.query(true).reply(200, { "data": { "repository": { "deployments": { "edges": deployments } } } });
+		};
+
+		const nockDeploymentListingApi = (deployments, repeatTimes) => {
+			"*".repeat(repeatTimes).split("").forEach(() => {
+				githubNock.get(`/repos/test-repo-owner/test-repo-name/deployments?environment=prod&per_page=10`)
+					.reply(200, deployments.map((item, idx) => ({
+						"id": item.node.databaseId,
+						"sha": item.node.commitOid,
+						"ref": "random",
+						"task": `task for deployment ${idx + 1}`,
+						"payload": {},
+						"original_environment": item.node.environment,
+						"environment": item.node.environment,
+						"description": `description for deployment ${idx + 1}`,
+						"creator": { "login": "test-repo-owner", "id": 1, "type": "User" },
+						"created_at": item.node.createdAt,
+						"updated_at": item.node.createdAt,
+						"statuses_url": "random",
+						"repository_url": "random",
+						"transient_environment": false,
+						"production_environment": true
+					})));
+			});
+		};
+
+		const nockDeploymentStatusApi = (deployments, repeatTimes) => {
+			"*".repeat(repeatTimes).split("").forEach(() => {
+				deployments.forEach((item, idx) => {
+					githubNock.get(`/repos/test-repo-owner/test-repo-name/deployments/${item.node.databaseId}/statuses?per_page=100`)
+						.reply(200, [
+							{ "id": idx * 1000 + 1, "state": "inactive", "description": "random", "environment": item.node.environment },
+							{ "id": idx * 1000 + 2, "state": "success", "description": "random", "environment": item.node.environment }
+						]);
+				});
+			});
+		};
+
+		const nockDeploymentCommitGetApi = (deployments, repeatTimes) => {
+			"*".repeat(repeatTimes).split("").forEach(() => {
+				deployments.forEach(item => {
+					githubNock.get(`/repos/test-repo-owner/test-repo-name/commits/${item.node.commitOid}`)
+						.reply(200, {
+							commit: {
+								author: { name: "random", email: "random", date: new Date() },
+								message: `commit message [JIRA-${item._seq}]`
+							}, html_url: "random"
+						});
+				});
+			});
+		};
+
+		const expectDeploymentEntryInDB = async (deployments) => {
+			for (const deployment of deployments) {
+				const key = createHashWithoutSharedSecret(`ghurl_${gitHubCloudConfig.baseUrl}_repo_${deployment.node.repository.id}_env_${deployment.node.environment}`);
+				const successStatusDate = deployment.node.statuses.nodes.find(n=>n.state === "SUCCESS")?.updatedAt;
+				const result = await ddb.getItem({
+					TableName: envVars.DYNAMO_DEPLOYMENT_HISTORY_CACHE_TABLE_NAME,
+					Key: {
+						"Id": { "S": key },
+						"CreatedAt": { "N": String(new Date(successStatusDate).getTime()) }
```

Comment:
Could probably move this out into a variable as well? `const createdAtTimestamp = successStatusDate && new Date(successStatusDate).getTime();` Then `"CreatedAt": { "N": String(createdAtTimestamp) }`?

Reply:
Same with the quotation marks too.

Reply:
fixed.